tests/js/object-tests.js

```diff
@@ -554,46 +554,55 @@ module.exports = {
         });
     },
 
-    async testNotification() {
-        const realm = new Realm({schema: [schemas.StringOnly]});
+    testNotification: async function () {
+        const realm = new Realm({ schema: [schemas.StringOnly] });
 
+        let calls = 0;
+        let resolve;
+        let reject;
+        
+        let promise = new Promise((res, rej) => { resolve = res, reject = rej });
```

Comment:
I don't like this pattern. Why not simply include the following (lines 565-592) in the callback passed to the `Promise` constructor?

Reply:
This test does strange things for sure. I just fixed the test. It's not a goal to refactor it, although I have debated myself if it would make sense to do that. I decided against it in this PR since it will get out of scope quickly